temp-legacy-code/src/main/java/com/ivy/legacy/datamodel/Tag.kt

```diff
@@ -0,0 +1,39 @@
+package com.ivy.legacy.datamodel
+
+import com.ivy.data.db.entity.TagEntity
+import java.time.LocalDateTime
+import java.util.UUID
+
+data class Tag(
+    val id: UUID = UUID.randomUUID(),
```

Comment:
Let's use typed-ids or i.e. `value class TagId` this way the compiler can prevent mistakes like functions interpreting this like some other type of id

Reply:
Seems overkill,
"functions interpreting this like some other type" can be classified as a dev error.

tag.id should be expressive enough to indicate its an id for tag

Reply:
It removes ambiguity from case like:
```kotlin
fun attachTagUnsafe(transactionId: UUID, tagId: UUID) { ... }
fun attachTag(transactionId: Transactionid, tagId: TagId) { ... }

fun main() {
   val transactionId = UUID.randomUUID()
   val tagId = UUID.randomUUID()
   tagTransactionUnsafe(tagId, transactionId) // compiles
   tagTransaction(tagId, transactionId) // does not compile
}
```

Also, value classes introduce no performance overhead but protect us as compile-time. Wdyt?